crates/worker/src/partition/shuffle.rs

```diff
@@ -223,12 +223,11 @@ where
         let state_machine = StateMachine::new(
             metadata,
             outbox_reader,
-            move |msg| {
-                let bifrost = bifrost.clone();
-                async move {
-                    append_envelope_to_bifrost(&bifrost, Arc::new(msg)).await?;
```

Comment:
I am not sure whether creating `Appenders` is so expensive. It looks that we might save a few atomic writes. Did you measure the impact of your changes? What could make a difference is the usage of the `BackgroundAppender` which also supports batching of appends. But this will require a bigger refactoring of the shuffler to make it work, I believe.